addons/stock/migrations/11.0.1.1/post-migration.py

```diff
@@ -253,6 +253,69 @@ def create_stock_move_line_reserved(env):
     )
 
 
+@openupgrade.logging()
+def create_stock_move_line_from_inventory_moves(env):
+    """This method creates stock.move.line got from inventory moves.
+    """
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO stock_move_line (
+            create_date,
+            create_uid,
+            date,
+            location_dest_id,
+            location_id,
+            lot_id,
+            lot_name,
+            move_id,
+            ordered_qty,
+            owner_id,
+            picking_id,
+            product_id,
+            product_qty,
+            product_uom_id,
+            product_uom_qty,
+            qty_done,
+            reference,
+            state,
+            write_date,
+            write_uid
+        )
+        SELECT
+            sm.create_date,
+            sm.create_uid,
+            sm.date,
+            sm.location_dest_id,
+            sm.location_id,
+            sil.prod_lot_id,
+            spl.name,
+            sm.id,
+            0,
+            sm.restrict_partner_id,
+            NULL,
+            sm.product_id,
+            0,
+            sm.product_uom,
+            0,
+            sm.product_uom_qty,
+            sm.name,
+            sm.state,
+            sm.write_date,
+            sm.write_uid
+        FROM stock_move sm
+        INNER JOIN stock_inventory si ON sm.inventory_id = si.id
+        LEFT JOIN (SELECT MIN(sil.id) AS id, sil.inventory_id,
```

Comment:
This is not going to work correctly, as only one line with lot will be found (if any) and assigned to all stock.move.line, instead of putting it to the proper one. You should do `LEFT JOIN` on the many2many table of the field `lot_ids` instead, as this one will contain the lot if any, and forget `stock.inventory.table`.

Reply:
field `lot_ids` is a compute, so there isn't many2many relation table

Reply:
Then you will need to dig in in how it's computed to get a similar result in this SQL, but current expression is not valid as stated.

Reply:
Digged and updated.